.github/workflows/main-ci-pipeline.yml

```
@@ -16,4 +16,28 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     - run: docker build --target test --tag todo-app:test .
-    - run: docker run todo-app:test
\ No newline at end of file
+    - run: docker run todo-app:test
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: |
+      github.ref == 'refs/heads/exercise-9' ||
+      (github.event_name == 'push' && github.ref == 'refs/heads/main')
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build Docker image
+        run: docker build --target production --platform linux/amd64 --tag ${{ secrets.DOCKER_USERNAME }}/todo-app:prod .
+      - name: Push Docker image to Docker Hub
+        run: docker push ${{ secrets.DOCKER_USERNAME }}/todo-app:prod
+      - name: Trigger Azure Deployment
+        run: |
+          curl -X POST ${{ secrets.AZURE_WEBHOOK_URL }}
```

I claim this command is silently failing (would expect a JSON payload response in the pipeline run). Can you see why?
```suggestion
          curl -X POST '${{ secrets.AZURE_WEBHOOK_URL }}'
```

---

Ah I see it's because there's a dollar sign in the webhook url. I should have explicitly checked. Thanks